pkg/proc/mem.go

```diff
@@ -179,7 +179,14 @@ func (mem *compositeMemory) WriteMemory(addr uint64, data []byte) (int, error) {
 
 			switch piece.Kind {
 			case op.RegPiece:
-				err := mem.regs.ChangeFunc(piece.Val, op.DwarfRegisterFromBytes(pieceMem))
+				newReg := op.DwarfRegisterFromBytes(pieceMem)
+				// Make sure the register is full-width, otherwise ChangeFunc could fail.
+				// Zero-extension is fine, because that's what most load instructions do,
+				// and this is effectively emulating a load to a register, not a store to
+				// memory.
+				newReg.Bytes = nil
```

Comment:
I'm worried about registers larger than 8 bytes, with this code. I think something like this would be better:

```
oreg := mem.regs.Reg(piece.Val)
b := pieceMem
if oreg != nil {
    if oreg.Bytes != nil {
        b = make([]byte, len(oreg.Bytes))
    } else {
        b = make([]byte, mem.arch.PtrSize())
    }
    //TODO(aarzilli): mix with original bytes from oreg if called through (*Eval).SetVariable
    copy(b, pieceMem)
}
newReg := op.DwarfRegisterFromBytes(b)
```

Reply:
I don't believe the compiler does any vectorization at this point, and there aren't any vector primitives, so I didn't really consider that. Do you just want to future-proof it?

I'm happy to make this change, just checking.

Reply:
> Do you just want to future-proof it?

Yes.